src/components/list-view/list-view.tsx

```diff
@@ -1,20 +1,86 @@
 import React, { PureComponent } from "react";
-import { View, Text } from "react-native";
+import { View, FlatList } from "react-native";
 import { styles } from "./style";
+import { EventCard } from "./event-card";
+import { data } from "../../../tst/mock-data";
+import { EventCardProps } from "./event-card/event-card";
 
 export interface ListViewProps {}
 
-export interface ListViewState {}
+export interface ListViewState {
+  refreshing: boolean;
+  events: EventCardProps[];
+}
 
 export class ListView extends PureComponent<ListViewProps, ListViewState> {
   constructor(props: ListViewProps) {
     super(props);
+
+    this.initializeEvents = this.initializeEvents.bind(this);
+    this.fetchMoreEvents = this.fetchMoreEvents.bind(this);
+    this.onRefresh = this.onRefresh.bind(this);
+
+    this.initializeEvents();
   }
 
+  initializeEvents() {
+    const events = this.fetchEventsFromDB();
+    this.state = {
+      refreshing: false,
+      events: events,
+    };
+  }
+
+  fetchEventsFromDB() {
+    console.log("fetching initial data");
+    // TODO: Replace with call from DB
+    return data;
+  }
+
+  fetchMoreEvents = () => {
+    console.log("reached end, fetching more data");
+    this.setState({ refreshing: true });
+
+    // TODO: Replace with call from DB
+    const moreEvents = this.fetchEventsFromDB();
+
+    this.setState({
+      refreshing: false,
+      events: [...this.state.events, ...moreEvents],
+    });
+  };
+
+  onRefresh = () => {
+    console.log("user has pulled to refresh");
+    this.setState({
+      events: this.fetchEventsFromDB(),
+    });
+  };
+
   render() {
     return (
       <View style={styles.container}>
-        <Text>TODO: Create List View!</Text>
+        <FlatList
+          data={this.state.events}
+          refreshing={this.state.refreshing}
+          onRefresh={this.onRefresh}
+          onEndReachedThreshold={1}
+          onEndReached={this.fetchMoreEvents}
+          keyExtractor={(item) => {
+            return item.id;
+          }}
+          renderItem={({ item }) => (
+            <EventCard
+              id={item.id}
```

Comment:
I've gotten several errors on console about the mock strings not being unique:
![image](https://user-images.githubusercontent.com/78049162/110837099-af4fb680-8255-11eb-9e01-2a7f95cb38ce.png)
Like that one, there is an identical error from `RandomStringId` to `RandomStringId16`.

Reply:
oh thats fine when the data is real it wont give out that warning. right now i init the list with data and the 12 events all have unique ids. then i scroll to the both and add data to it again so every id is doubled. but since this data is just for testing the warning can be ignored.